docs/program-setup/bare_metal/baremetal_usage.md

```diff
@@ -0,0 +1,75 @@
+# Using bare metal profile
+
+## Creating a bare metal application
+
+First, clone Mbed OS into the directory where we want to create an application
+```
+mkdir my_mbed_app && cd my_mbed_app
+git clone https://github.com/ARMmbed/mbed-os.git
+```
+
+To enable Mbed OS bare metal, create an `mbed_app.json` with the following contents:
+```
+{
+    "requires": ["bare-metal"]
+}
+```
+
+On bare metal, features are enabled with a bottom-up approach - those listed in [Bare metal APIs](bare_metal_api.md#bare-metal-apis) are now ready for use (but not other features) in our application. See [Bare metal example](baremetal_example.md).
+
+To pull in additional features (take filesystem for example)
```

Comment:
As discussed, FileSystem is quite complex for simple IOT devices. Let's use an event queue instead.